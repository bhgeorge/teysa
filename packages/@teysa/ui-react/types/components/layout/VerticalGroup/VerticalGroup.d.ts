import { ReactNode } from 'react';
declare type VerticalGroupSize = 'sm' | 'md' | 'lg';
interface VerticalGroupProps {
    size?: VerticalGroupSize;
    children: ReactNode | ReactNode[];
}
declare const VerticalGroup: ({ size, children }: VerticalGroupProps) => JSX.Element;
export default VerticalGroup;
