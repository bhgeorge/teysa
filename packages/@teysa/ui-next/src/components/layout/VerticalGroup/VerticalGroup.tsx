// Global
import React, { ReactNode } from 'react';
import classNames from 'classnames';

type VerticalGroupSize = 'sm' | 'md' | 'lg';

interface VerticalGroupProps {
  size?: VerticalGroupSize;
  children: ReactNode | ReactNode[];
}

const sizeClasses: Record<VerticalGroupSize, string> = {
  sm: 'gap-y-8',
  md: 'gap-y-16',
  lg: 'gap-y-32',
};

export function VerticalGroup({ size = 'md', children }: VerticalGroupProps) {
  return (
    <div className={classNames('grid', sizeClasses[size])}>{children}</div>
  );
}
