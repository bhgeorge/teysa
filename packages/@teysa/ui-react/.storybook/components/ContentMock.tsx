// Global
import React from 'react';

const ContentMock = () => (
  <div className="p-8 flex items-center justify-center w-full bg-theme-text text-theme-bg">
    content
  </div>
);

export default ContentMock;
