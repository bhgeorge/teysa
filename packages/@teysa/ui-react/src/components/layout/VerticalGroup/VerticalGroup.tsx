// Global
import React, { ReactNode } from 'react';
import classNames from 'classnames';

type VerticalGroupSize = 'sm' | 'md' | 'lg';

interface VerticalGroupProps {
  size?: VerticalGroupSize;
  children: ReactNode | ReactNode[];
}

const sizeClasses: Record<VerticalGroupSize, string> = {
  sm: 'gap-y-8',
  md: 'gap-y-16',
  lg: 'gap-y-32',
};

const VerticalGroup = ({ size = 'md', children }: VerticalGroupProps) => (
  <div className={classNames('grid', sizeClasses[size])}>{children}</div>
);

export default VerticalGroup;
