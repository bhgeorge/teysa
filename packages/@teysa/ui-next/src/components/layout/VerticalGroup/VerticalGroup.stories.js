// Storybook
import ContentMock from '../../../../.storybook/components/ContentMock';
// Local
import { VerticalGroup } from './VerticalGroup';

export default {
  title: 'Layout/VerticalGroup',
  component: VerticalGroup,
};

const Template = props => (
  <VerticalGroup {...props}>
    {[0, 1, 2].map(i => (
      <ContentMock key={i} />
    ))}
  </VerticalGroup>
);

export const Default = Template.bind({});
