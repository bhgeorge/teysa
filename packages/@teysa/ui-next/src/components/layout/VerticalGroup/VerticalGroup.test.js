// Global
import { render } from '@testing-library/react';
import React from 'react';
// Storybook
import ContentMock from '../../../../.storybook/components/ContentMock';
// Local
import { VerticalGroup } from './VerticalGroup';

const snapshot = size => {
  const component = render(
    <VerticalGroup size={size}>
      <ContentMock />
    </VerticalGroup>
  );

  expect(component.container.firstChild).toMatchSnapshot();
};

it('renders correctly at all sizes', () => {
  snapshot('sm');
  snapshot('md');
  snapshot('lg');
});
