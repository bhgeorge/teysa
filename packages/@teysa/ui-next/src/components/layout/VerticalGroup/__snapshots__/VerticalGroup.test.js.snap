// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly at all sizes 1`] = `
<div
  class="grid gap-y-8"
>
  <div
    class="p-8 flex items-center justify-center w-full bg-theme-text text-theme-bg"
  >
    content
  </div>
</div>
`;

exports[`renders correctly at all sizes 2`] = `
<div
  class="grid gap-y-16"
>
  <div
    class="p-8 flex items-center justify-center w-full bg-theme-text text-theme-bg"
  >
    content
  </div>
</div>
`;

exports[`renders correctly at all sizes 3`] = `
<div
  class="grid gap-y-32"
>
  <div
    class="p-8 flex items-center justify-center w-full bg-theme-text text-theme-bg"
  >
    content
  </div>
</div>
`;
